rust-examples/enum.rs

```diff
@@ -0,0 +1,63 @@
+#![feature(fundamental, lang_items, no_core)]
+#![allow(dead_code)]
+#![no_core]
+
+enum Tag {
+    A(i32),
+    B(i32),
+    C,
+}
+
+enum CTag {
+    A = -1,
+    B = 0,
+    C = 1,
+    D = 12,
+}
+
+fn main() {
+    let t = Tag::B(17);
+    let i = match t {
+        Tag::A(_) => 1,
+        Tag::B(i) => i,
+        _ => 3
+    };
+    wasm::print_i32(i); // (i32.const 17)
+
+    let t = CTag::A;
+    let i = match t {
+        i @ _ => i as i32
+    };
+    wasm::print_i32(i as i32); // (i32.const -1)
+    wasm::print_i32(CTag::D as i32); // (i32.const 12)
+}
+
+pub mod marker {
+    use clone::Clone;
+
+    #[lang = "sized"]
+    #[fundamental]
+    pub trait Sized { }
+
+    #[lang = "copy"]
+    pub trait Copy : Clone { }
+}
+
+pub mod clone {
```

Comment:
Do cross-file imports work yet? If so, it'd be good to share the implementation of these traits. I realize in the long run these will probably be handled by libstd, but it may be good to avoid some duplication in the short term. Feel free to save this for a future PR, or ignore entirely :)